model/qti/ImportService.php

```diff
@@ -883,28 +880,33 @@ protected function getMetadataImporter()
 
     /**
      * Retrieve the labels of all parent classes up to base item class.
-     *
-     * Return values: an empty string if the $class parameter is not a class object
-     *                an empty array if the $class parameter is the base item class
-     *                an array in other cases, ordered from base item class to the class given as parameter
-     *
-     * @param mixed $class Class from which to retrieve.
-     * @return array|string
      */
-    public function retrieveFullPathLabels($class)
-    {
-        if (!$class instanceof core_kernel_classes_Class) {
-            return '';
+    public function getTargetClassForAssets(
+        core_kernel_classes_Class $itemClass,
+        core_kernel_classes_Resource $itemResource
+    ): core_kernel_classes_Class {
+        // Collecting labels path from item root to the class where the item resource is stored
+        $labels = [];
+        while ($itemClass->getUri() !== TaoOntology::CLASS_URI_ITEM) {
```

Comment:
I'm worried that we could end here with infinite loop? I don't know if class could reference to itself or something?

Reply:
it's the very old code which has never cause any problem, covered by tests, if we had RDF tree broken it here would be a last problem we should be worried about.